### PR TITLE
Kill RIOSection for good##refactor

### PR DESCRIPTION
**Pull request by ret2libc**

### CODE REVIEW

File: libr/bin/p/bin_elf.inc [hunk 1]
```diff
@@ -877,22 +877,24 @@ static RList* patch_relocs(RBin *b) {
...
 	if (!g) {
 		return NULL;
 	}
-	n_off = g->paddr + g->size;
-	n_vaddr = g->vaddr + g->vsize;
+	n_vaddr = g->itv.addr + g->itv.size;
 	//reserve at least that space
 	size = bin->reloc_num * 4;
-	if (!b->iob.section_add (io, n_off, n_vaddr, size, size, R_PERM_R, ".got.r2", 0, io->desc->fd)) {
+	RIOMap *gotr2map = b->iob.map_add (io, io->desc->fd, R_PERM_R, 0, n_vaddr, size);
```
Comment (ret2libc) [inline libr/bin/p/bin_elf.inc]: a `null://` map should be added here, not the `io->desc->fd`.

**Comment by ret2libc** [comment]
@radare this is ready to be reviewed!

File: libr/bin/p/bin_elf.inc [hunk 2]
```diff
@@ -877,22 +877,31 @@ static RList* patch_relocs(RBin *b) {
...
+	free (muri);
+	if (!gotr2desc) {
 		return NULL;
 	}
+
+	RIOMap *gotr2map = b->iob.map_get (io, n_vaddr);
+	if (!gotr2map) {
+		return NULL;
+	}
+	gotr2map->name = strdup (".got.r2");
```
Comment (radare) [inline libr/bin/p/bin_elf.inc]: fishy

Comment (ret2libc) [reply libr/bin/p/bin_elf.inc]: in which way? this was already like this. It is to create the "fake" got section used to patch relocations.

**Pull request merged**